### PR TITLE
tox

### PR DESCRIPTION
**Pull request by steenzout**
- added tox support
- updated `.gitignore` rules
- changed lambda for function as recommended by `py.test`
- updated travis-ci tests
  - parallel
  - `tox` usage

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,109 @@
...
+.Python
+[Bb]in
+[Ii]nclude
+[Ll]ib
+[Ll]ib64
+[Ll]ocal
+[Ss]cripts
+pyvenv.cfg
+.venv
+pip-selfcheck.json
```
Comment (vitalk) [inline .gitignore]: This `.gitignore` contains a lot of useless stuff, e.g. scrapy, django, flask, translations, etc. I prefer to keep it as small as possible. Please, add only required lines to it.

File: .travis.yml [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
+    - "${HOME}/virtualenv"
+    - "${TRAVIS_BUILD_DIR}/.tox"
+
 python:
   - 2.6
   - 2.7
   - 3.3
   - 3.4
+  - 3.5
+  - 3.6
```
Comment (vitalk) [inline .travis.yml]: Extension is not tested for python 3.5 and 3.6 environments and has failed build on Travis. Allow failures for python 3.5 as well.

**Comment by vitalk** [comment]
Hello @steenzout! Thank you for your PR. Please fix noted issues to ensure the build is successful.

File: .travis.yml (hunk 2 above)
Comment (steenzout) [inline .travis.yml]: done.

File: .gitignore (hunk 1 above)
Comment (steenzout) [inline .gitignore]: done.

**Comment by steenzout** [comment]
@vitalk done.

**Comment by steenzout** [comment]
@vitalk anything else?

**Comment by vitalk** [comment]
Sorry for delay and thank you!

**Pull request merged**